### PR TITLE
[Process] remove deprecated features

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
tests are failing :)

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -1538,7 +1393,7 @@ private function close()
             if ($this->processInformation['signaled'] && 0 < $this->processInformation['termsig']) {
                 // if process has been signaled, no exitcode but a valid termsig, apply Unix convention
                 $this->exitcode = 128 + $this->processInformation['termsig'];
-            } elseif ($this->enhanceSigchildCompatibility && $this->isSigchildEnabled()) {
+            } else  {
```
Comment (fabpot) [inline src/Symfony/Component/Process/Process.php]: extra space here

**Review by fabpot — CHANGES_REQUESTED** [review]
Tests still do not pass apparently.

File: src/Symfony/Component/Process/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+ * added a second `array $env = array()` argument to the `start()` method of
+   the `Process` class
+ * the `areEnvironmentVariablesInherited()`, `getOptions()`, `setOptions()`,
+   `getEnhanceWindowsCompatibility()`, `getEnhanceSigchildCompatibility()`,
+   `setEnhanceSigchildCompatibility()`, and `setEnhanceWindowsCompatibility()`
+   methods of the `Process` class have been removed
+ * environment variables will always be inherited
+ * the `ProcessUtils::escapeArgument()` method has been removed
+ * the `inheritEnvironmentVariables()` and `setOption()` options of the
+   `ProcessBuilder` class have been removed
```
Comment (stof) [inline src/Symfony/Component/Process/CHANGELOG.md]: Wrong changelog entry. The whole class is gone

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by xabbuh** [comment]
I have removed the sigchild compatibility part as I did not manage to make it work properly. If someone wants to give it a try, you can just base your work on this PR or create a new one.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Let's merge this one as is then.

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**